### PR TITLE
add Global Exeption

### PR DESCRIPTION
**Pull request by YuraNR**

### CODE REVIEW

**Review by KDanil2000 — CHANGES_REQUESTED** [review]
I opened the "Pagination, Sorting, Swagger" task to check, but here I see the "GlobalExceptionHandler" task. Please download the correct link)

File: pom.xml [hunk 1]
```diff
@@ -69,7 +67,10 @@
             <groupId>org.liquibase</groupId>
             <artifactId>liquibase-core</artifactId>
         </dependency>
-
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
+        </dependency>
```
Comment (liliia-ponomarenko) [inline pom.xml:73]: better use 
```
 <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-validation</artifactId>
 </dependency>
```

File: src/main/java/bookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,11 +1,18 @@
 package bookstore.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 
 public record CreateBookRequestDto(Long id,
+                                   @NotNull
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/dto/CreateBookRequestDto.java:8]: with String type better use NotBlank

File: src/main/java/bookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
                                    String title,
+                                   @NotNull
                                    String author,
+                                   @NotNull
                                    String isbn,
+                                   @NotNull
+                                   @Min(1)
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/dto/CreateBookRequestDto.java:15]: Better use PositiveOrZero instead  Min(1)

File: src/main/java/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -48,7 +49,7 @@ public void deleteById(@PathVariable Long id) {
 
     @ResponseStatus(HttpStatus.OK)
     @PutMapping("/{id}")
-    public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestDto
+    public BookDto updateBook(@PathVariable Long id, @RequestBody @Valid CreateBookRequestDto
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/controller/BookController.java:52]: just FYI

you may also validate such things as @PathVariable, @RequestParams etc. It looks like:

`@PathVariable @Positive Long id`
In order to be able to perform such validation you need to annotate your controller class with Validated annotation

again, this is just FYI - apply it or not :)

File: src/main/java/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/exception/CustomGlobalExceptionHandler.java:29]: time is pretty useless in response body - it does not bring any value to the client

status is obvious as well - no sense to have it in body

File: src/main/java/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/exception/CustomGlobalExceptionHandler.java:21]: do you have any other exceptions except for MethodArgumentNotValidException?

does it make sense to handle them as well? for example  EntityNotFoundException

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution;)

File: src/main/java/bookstore/controller/BookController.java [hunk 7]
```diff
@@ -36,7 +39,7 @@ public BookDto getBookById(@PathVariable Long id) {
 
     @ResponseStatus(HttpStatus.CREATED)
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/controller/BookController.java:42]: Анотація Validated над класом потрібна для того якщо ти хочеш валідувати id, але вона не заміняє анотацію Valid для тіла запиту, тому давай виправимо твій метод оновлення сутності 
```
@PutMapping("/{id}")
    public BookDto updateBook(@PathVariable @Positive Long id, @RequestBody @Valid CreateBookRequestDto
            requestDto) {
        return bookService.updateById(id, requestDto);
    }
```

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! But fix my comment before merging ;)

**Pull request merged**